### DISCUSSION TITLE
Understanding the point of device types?

### DISCUSSION THREAD

**Discussion by deanfourie1** [netbox-community/netbox#13759] [General]
So, i'm a little confused with device types.

Right now, if I have say 10x 48 port PoE switches to add, i'm having to add each one individually and then go in and create 48 interfaces and assign them to each device. I'm no sure if i'm doing this wrong but,

Wouldnt it be better to create a device, say 48 Port PoE switch and then add 48 interfaces to that.

Then I can come along and add a Aruba 48 port switch, a cisco 48 port switch and a Mikrotik 48 port switch, and they will all have 48 interfaces automatically assigned to them?

I feel like i'm doing this the hard way.

Thanks

**Comment by candlerb** [comment]
You should create:
* A Device Type for the Aruba 48 port switch (specific model)
* A Device Type for the Cisco 48 port switch (specific model)
* A Device Type for the Mikrotik 48 port switch (specific model)

For each of these Device Types, you would add 48 interfaces - and anything else needed, e.g. console ports, power ports.

Then when you create an instance of one of these devices, it will get a copy of all the interfaces defined in the Device Type.

Note that if you change the Device Type, e.g. adding or removing components, it *won't* affect any existing Devices of that Type.

**Reply by deanfourie1** [reply]
Ok thanks, just wanted to make sure I was on the right track or it would be a lot of wasted time haha.

**Reply by candlerb** [reply]
Also note there are lots of pre-defined device types you can install here: https://github.com/netbox-community/devicetype-library

Plenty of Cisco and Mikrotik models. For Aruba see the HPE directory.
